pkg/services/librarypanels/api.go

```diff
@@ -0,0 +1,31 @@
+package librarypanels
+
+import (
+	"github.com/go-macaron/binding"
+	"github.com/grafana/grafana/pkg/api"
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/middleware"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+func (lps *LibraryPanelService) registerAPIEndpoints() {
+	if !lps.IsEnabled() {
+		return
+	}
+
+	lps.RouteRegister.Group("/api/library-panels", func(libraryPanels routing.RouteRegister) {
+		libraryPanels.Post("/", middleware.ReqSignedIn, binding.Bind(AddLibraryPanelCommand{}), api.Wrap(lps.createHandler))
+	})
+}
+
+// createHandler handles POST /api/library-panels.
+func (lps *LibraryPanelService) createHandler(c *models.ReqContext, cmd AddLibraryPanelCommand) api.Response {
+	panel, err := lps.Create(c, cmd)
+
+	if err != nil {
+		return api.Error(500, "Failed to create library panel", err)
+	}
+
+	return api.JSON(200, util.DynMap{"panel": panel})
+}
```

Comment:
I added __`createHandler`__ because I wanted to separate the Create to a service level interface without the api response and I couldn't find a good way to test the resulting response status. Any suggestions?

Reply:
What's the actual problem you're trying to solve @hugohaggmark, that you can't test the response in the test?

Reply:
I thought the LibraryPanelService interface should be separate from the Api, so if other services consume this service it would become clearer, but maybe there is a lot I don't understand with the backend architecture (yet). 

How do services interact with each other? This service will need to interact with both dashboards, tags and folders moving forward.

Reply:
Think separating DB create action into a separate service function seems like a very good pattern to me.